smartsim/database/orchestrator.py

```diff
@@ -420,6 +431,7 @@ def set_hosts(self, host_list: t.List[str]) -> None:
             raise TypeError("host_list argument must be a list of strings")
         if not all(isinstance(host, str) for host in host_list):
             raise TypeError("host_list argument must be list of strings")
+        self._user_hostlist = host_list
```

Comment:
Quick (and potentially overcautious) warning here:

This is saving a reference and not a copy. This code fragment will raise an exception at the assert stmt
```py
hosts = ["host1", "host2", "host3"]
orc = Orcestrator(hosts=hosts, ...)
hosts.append("host4")
assert len(orc._user_hostlist) == 3  # <-- unexpected error here
```

Reply:
Good catch, let's make a copy!